src/Features/Core/Diagnostics/AnalyzerHelper.cs

```diff
@@ -34,43 +34,36 @@ public static bool IsCompilerAnalyzer(DiagnosticAnalyzer analyzer)
-        {
-            return diagnostic =>
-                hostDiagnosticUpdateSource?.ReportAnalyzerDiagnostic(analyzer, diagnostic, workspace, null);
-        }
-
-        public static AnalyzerExecutor GetAnalyzerExecutorForSupportedDiagnostics(
+        internal static AnalyzerExecutor GetAnalyzerExecutorForSupportedDiagnostics(
             DiagnosticAnalyzer analyzer,
             AbstractHostDiagnosticUpdateSource hostDiagnosticUpdateSource,
-            Func<Exception, DiagnosticAnalyzer, bool> continueOnAnalyzerException,
-            CancellationToken cancellationToken)
+            Action<Exception, DiagnosticAnalyzer, Diagnostic> onAnalyzerException = null,
+            CancellationToken cancellationToken = default(CancellationToken))
         {
-            var addExceptionDiagnostic = GetAddExceptionDiagnosticDelegate(analyzer, hostDiagnosticUpdateSource, hostDiagnosticUpdateSource?.Workspace);
-
             // Skip telemetry logging if the exception is thrown as we are computing supported diagnostics and
             // we can't determine if any descriptors support getting telemetry without having the descriptors.
-            return AnalyzerExecutor.CreateForSupportedDiagnostics(addExceptionDiagnostic, continueOnAnalyzerException, cancellationToken);
+            Action<Exception, DiagnosticAnalyzer, Diagnostic> defaultOnAnalyzerException = (ex, a, diagnostic) =>
+                OnAnalyzerException_NoTelemetryLogging(ex, a, diagnostic, hostDiagnosticUpdateSource);
+
+            return AnalyzerExecutor.CreateForSupportedDiagnostics(onAnalyzerException ?? defaultOnAnalyzerException, cancellationToken);
         }
 
-        public static AnalyzerExecutor GetAnalyzerExecutor(
-            DiagnosticAnalyzer analyzer,
+        internal static void OnAnalyzerException_NoTelemetryLogging(
+            Exception e,
+            DiagnosticAnalyzer analyzer, 
+            Diagnostic diagnostic,
             AbstractHostDiagnosticUpdateSource hostDiagnosticUpdateSource,
-            Project project,
-            Compilation compilation,
-            Action<Diagnostic> addDiagnostic,
-            AnalyzerOptions analyzerOptions,
-            Func<Exception, DiagnosticAnalyzer, bool> continueOnAnalyzerException,
-            CancellationToken cancellationToken)
+            Project projectOpt = null)
+        {
+            if (diagnostic != null)
         {
-            var addExceptionDiagnostic = GetAddExceptionDiagnosticDelegate(analyzer, hostDiagnosticUpdateSource, project);
-            return AnalyzerExecutor.Create(compilation, analyzerOptions, addDiagnostic, addExceptionDiagnostic, continueOnAnalyzerException, cancellationToken);
+                hostDiagnosticUpdateSource?.ReportAnalyzerDiagnostic(analyzer, diagnostic, hostDiagnosticUpdateSource?.Workspace, projectOpt);
+        }
+        
+            if (IsBuiltInAnalyzer(analyzer))
+        {
```

Comment:
Yes, it seems merge caused quite a few spacing/indent issues. I'll fix all of these.